pkg/client/cache/controller.go

```diff
@@ -61,21 +61,23 @@ type Config struct {
 type ProcessFunc func(obj interface{}) error
 
 // Controller is a generic controller framework.
-type Controller struct {
+type controller struct {
 	config         Config
 	reflector      *Reflector
 	reflectorMutex sync.RWMutex
 }
 
 // TODO make the "Controller" private, and convert all references to use ControllerInterface instead
-type ControllerInterface interface {
+type Controller interface {
 	Run(stopCh <-chan struct{})
 	HasSynced() bool
+	LastSyncResourceVersion() string
+	Requeue(obj interface{}) error
```

Comment:
In `func (s *sharedIndexInformer) LastSyncResourceVersion() {...}`, it uses `controller.reflector.LastSyncResourceVersion()`; so I expose it here.